src/mjolnir/ingest_transit.cc

```diff
@@ -616,28 +622,37 @@ void ingest_tiles(const boost::property_tree::ptree& pt,
     queue.pop();
     uniques.lock.unlock();
 
+    // PBF objects throw when they're too big, so keep writing .pbf.x tiles until there's no more
     Transit tile;
-    auto file_name = GraphTile::FileSuffix(current.graphid, SUFFIX_NON_COMPRESSED);
-    file_name = file_name.substr(0, file_name.size() - 3) + "pbf";
-    filesystem::path transit_tile = pt.get<std::string>("mjolnir.transit_dir") +
-                                    filesystem::path::preferred_separator + file_name;
+    bool dangles = false;
+    uint16_t ext = 0;
 
-    // tiles are wrote out with .pbf or .pbf.n ext
-    const std::string& prefix = transit_tile.string();
-    LOG_INFO("Writing " + prefix);
+    const auto tile_path = get_tile_path(pt.get<std::string>("mjolnir.transit_dir"), current.graphid);
+    // first tile also gets an extension
+    auto current_path = tile_path;
+    do {
+      std::unordered_map<feed_object_t, GraphId> platform_node_ids = write_stops(tile, current);
+      dangles = write_stop_pairs(tile, current, platform_node_ids, uniques) || dangles;
+      write_routes(tile, current);
+      write_shapes(tile, current);
+
+      if (tile.stop_pairs_size() > 500000) {
+        LOG_INFO("Writing " + current_path);
+        write_pbf(tile, current_path);
+        tile.Clear();
+        current_path = tile_path + "." + std::to_string(ext++);
+      }
 
-    std::unordered_map<feed_object_t, GraphId> platform_node_ids = write_stops(tile, current);
-    bool dangles = write_stop_pairs(tile, current, platform_node_ids, uniques);
-    write_routes(tile, current);
-    write_shapes(tile, current);
+    } while (filesystem::exists(current_path));
```

Comment:
hmpf, this was too quick: it'll be a forever loop once a tile has more than 500k stop pairs, since I didn't write any code that removes any objects from the current `transit_info`.. will look at it tmrw again..